src/shared/filesystem.ts

```diff
@@ -6,102 +6,50 @@
 'use strict'
 
 import * as fs from 'fs'
+import * as os from 'os'
+import * as path from 'path'
+import { promisify } from 'util'
 
-export async function accessAsync(path: string | Buffer): Promise<void> {
-    await new Promise<void>((resolve, reject) => fs.access(path, err => {
-        if (!err) {
-            resolve()
-        } else {
-            reject(err)
-        }
-    }))
+const access = promisify(fs.access)
+
+export type PathLike = fs.PathLike
+
+export const accessAsync = async (pathLike: fs.PathLike): Promise<void> => {
+    return access(pathLike)
 }
 
-export async function mkdirAsync(
-    path: fs.PathLike,
-    options?: number | string | fs.MakeDirectoryOptions | undefined | null
-): Promise<void> {
-    await new Promise<void>((resolve, reject) => fs.mkdir(path, options, err => {
-        if (!err) {
-            resolve()
-        } else {
-            reject(err)
-        }
-    }))
+export const mkdirAsync = promisify(fs.mkdir)
+
+const mkdtemp = promisify(fs.mkdtemp)
+
+export const getTempDirPath = (prefix: string = 'vsctk') => {
+    return path.join(
+        os.type() === 'Darwin' ? '/tmp' : os.tmpdir(),
+        prefix || 'vsctk'
+    )
 }
 
-export async function mkdtempAsync(prefix: string): Promise<string> {
-    return await new Promise<string>((resolve, reject) => {
-        fs.mkdtemp(prefix, (err, folder) => {
-            if (!err) {
-                resolve(folder)
-            } else {
-                reject(err)
-            }
-        })
-    })
+export const  mkdtempAsync = async (prefix?: string) => {
+    return mkdtemp(getTempDirPath(prefix))
```

Comment:
This file is intended to be a pure async wrapper around the fs methods used. Your change overrides the functionality, conditionally forcing the top level location. What if we moved this implementation to `filesystemUtilities.ts` (and gave it a suitable method name in the process), and did not export mkdtempAsync in here, so that we continue to use this overriding behavior? This keeps the door open to adding a pure mkdtempAsync call in the future, if we find a need for it.

Reply:
I don't know. I'm not really a fan of having both `filesystem` and `filesystemUtilities`. They are both convenience wrappers/utilities around `fs`. Ideally one of these two will go away... especially as we move to using the [fs_promises_api](https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api).

If we need to use `fs`, we can it's not that hard. I'd like to resist creating code "we might need to use" a.k.a. "dead code".